src/core/Table/EditableCell.tsx

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { Cell, TableCellProps } from 'react-table';
+
+export type OnCellEditCallbackParams<T extends Record<string, unknown>> = {
+  columnId: string;
+  value: string;
+  rowData: T;
+};
+
+export type EditableCellProps<T extends Record<string, unknown>> = {
+  cell: Cell<T>;
+  cellProps: TableCellProps;
+  onCellEdit?: (cellEditParams: OnCellEditCallbackParams<T>) => void;
+  children: React.ReactNode;
+};
+
+const EditableCell = <T extends Record<string, unknown>>(
+  props: EditableCellProps<T>,
+) => {
+  const { cell, cellProps, onCellEdit, children } = props;
+
+  const [value, setValue] = React.useState(cell.value);
+  const [isDirty, setIsDirty] = React.useState(false);
+
+  return (
+    <div
+      {...cellProps}
+      contentEditable
+      suppressContentEditableWarning
+      onInput={(e) => {
+        setValue(
+          (e.target as HTMLElement).innerText.replace(/(\r\n|\n|\r)/gm, ''),
+        );
+        setIsDirty(true);
+      }}
+      onBlur={() =>
+        isDirty &&
+        onCellEdit?.({
```

Comment:
So I was thinking that we might handle the cell editing internally, i.e. update table data on blur.

But that might be more complicated and less flexible, which makes me wonder... Why not just provide this `EditableCell` component (with styling) for the user to pass into `Cell` or `defaultCell`? Or is there some data/state that the user may not have access to?

Reply:
Yeah, I think `cellRenderer` would be a better approach than adding all these extra props. It would also future proof us for other kinds of cells (e.g. status cell).